homework-classes/App.js

```diff
@@ -1,78 +1,100 @@
 'use strict';
 
 /* global Util, Repository, Contributor */
+const root = document.getElementById('root');
+const HYF_REPOS_URL = 'https://api.github.com/orgs/HackYourFuture/repos?per_page=100';
 
 class App {
   constructor(url) {
-    this.mainContainer = null;
-    this.initialize(url);
+    this.createHeader(url);
+  }
+
+  createHeader(url) {
+    const header = Util.createAndAppend('header', root, {
+      class: 'container jumbotron mt-3 mb-3 px-5',
+    });
+    Util.createAndAppend('main', root, {
+      id: 'container',
+      class: 'container align-items-start',
+    });
+    const headerDiv = Util.createAndAppend('div', header, {
+      class: 'header-group d-flex flex-row justify-content-between mb-5 px-5',
+    });
+    Util.createAndAppend('h2', headerDiv, {
+      text: 'Hack Your Future Repositories',
+      class: 'display-4 d-flex align-items-center mx-auto col-sm header-title',
+    });
+    const hyfLink = Util.createAndAppend('a', headerDiv, {
+      href: 'https://www.hackyourfuture.net/',
+      target: '_blank',
+    });
+    Util.createAndAppend('img', hyfLink, {
+      class: 'img-fluid rounded img-thumbnail mx-auto',
+      src: './hyf.png',
+      alt: 'hack your future thumbnail',
+    });
+    Util.createAndAppend('select', header, {
+      id: 'repo-select',
+      class: 'form-control',
+      'aria-label': 'Hack Your Future Repositories Selection',
+    });
+    this.fetchRepositories(url);
   }
 
   /**
-   * Initialization
    * @param {string} url The GitHub URL for obtaining the organization's repositories.
    */
-  async initialize(url) {
-    // Add code here to initialize your app
-    // 1. Create the fixed HTML elements of your page
-    // 2. Make an initial XMLHttpRequest using Util.fetchJSON() to populate your <select> element
-
-    const root = document.getElementById('root');
-    const header = Util.createAndAppend('header', root, { class: 'header' });
-    this.mainContainer = Util.createAndAppend('div', root, { id: 'container' });
 
```

Comment:
When you create Repository and  Contributor objects by calling their respective constructors, you pass the relevant object obtained from the fetch request as an argument. From that point on that data is assumed 'owned' by the relevant class instance (= object created with `new`). No other objects should access that data directly. The owning object can define methods to provide access to some properties (an example in the starter code would be the `.name()` method in the Repository class), but it remains the owning object who remains in control. True OOP languages have the ability to enforce this by making class data 'private', i.e. only accessible to the owning object. Very recent version of JavaScript have this too, but it is too new to consider using.

So once you have created an array of Repository objects from the fetch data, all 'behavior' that invoved the repository data should be handled by the Repository object. When sorting the Repository object you need access to the `name`. You can ask the Repository object for the name by calling its `.name()` method.

The Repository class is also the best 'actor' to handle fetching the corresponding Contributors, because it has access to the `contributor_url`. The App class should not be concerned with this, it should just deal with the overall orchestration.